src/Storage/FileStorage.php

```diff
@@ -126,7 +126,10 @@ public function clear(): void
      */
     private function collectIndexData(): array
     {
-        $indexData = ['id' => $this->idGenerator->getId()];
+        $indexData = [
+            'id' => $this->idGenerator->getId(),
+            'collectors' => array_keys($this->collectors),
```

Comment:
Maybe collect keys in the loop below?

Reply:
Why?

Reply:
For performance, likely gain is significant though.

Reply:
I don't think there are much difference, but the first option looks better for me. 
Would you like to test? :)

Reply:
It's optional, can be left as is. By the way how big the array size can be?

Reply:
Now it has 11 elements